compiler/il/OMRNode.cpp

```diff
@@ -2357,7 +2357,101 @@ OMR::Node::isNotCollected()
    return false;
    }
 
+/**
+ * \brief
+ *    Check if the node is an internal pointer.
+ *
+ * \return
+ *    Return true if the node is an internal pointer, false otherwise.
+ *
+ * \note
+ *    This query currently only works on opcodes that can have a pinning array pointer,
+ *    i.e. aiadd, aiuadd, aladd, aluadd. Suppport for other opcodes can be added if necessary.
+ */
+bool
+OMR::Node::computeIsInternalPointer()
+   {
+   if (self()->getOpCode().hasPinningArrayPointer())
```

Comment:
On second thought, given the restriction in the doc comment, maybe this should be an assertion forcing callers to check. Currently callers assume that `computeIsInternalPointer()` implies `hasPinningArrayPointer()`, which gets in the way of later returning true for other opcodes, e.g. for `ternary` of two `aladd`s, or for `aload` of a pinning array temp